### PR TITLE
Add V2 payload handler to decrypt request body before calling endpoin…

### PR DESCRIPTION
**Pull request by haitaottd**
…t handler and encrypt response body before returning to client

### CODE REVIEW

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 1]
```diff
@@ -179,13 +191,30 @@ private Router createRoutesSetup() throws IOException {
...
 
+        setupV2Routes(router);
+
         return router;
     }
 
+    private void setupV2Routes(Router mainRouter) {
+        final Router v2Router = Router.router(vertx);
+
+        v2Router.post("/token/validate").handler(auth.handleV2(
```
[the remark this answers is not in the document]
Comment (haitaottd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:202]: `generate` and `refresh` in next PR. Will take a look at logout

File: src/main/java/com/uid2/operator/vertx/V2PayloadHandler.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        this.clock = clock;
+    }
+
+    public void handle(RoutingContext rc, Handler<RoutingContext> apiHandler) {
+        ClientKey ck = AuthMiddleware.getAuthClient(ClientKey.class, rc);
+
+        // Encrypted body format:
+        //  byte 1: version
+        //  byte 2-16: IV
+        //  byte 17-end: encrypted payload
```
[the remark this answers is not in the document]
Comment (haitaottd) [reply src/main/java/com/uid2/operator/vertx/V2PayloadHandler.java:44]: it's embedded in encrypted payload

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 3]
```diff
@@ -430,6 +494,37 @@ private void handleBucketsV1(RoutingContext rc) {
...
+        final JsonObject req = (JsonObject) rc.data().get("request");
+        final String qp = req.getString("since_timestamp");
+
+        if (qp != null) {
+            final Instant sinceTimestamp;
+            try {
+                LocalDateTime ld = LocalDateTime.parse(qp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+                sinceTimestamp = ld.toInstant(ZoneOffset.UTC);
+            } catch (Exception e) {
+                ResponseUtil.ClientError(rc, "invalid date, must conform to ISO 8601");
```
[the remark this answers is not in the document]
Comment (haitaottd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:507]: I think that makes error handling difficult. Client has to differentiate errors created by service, which are encrypted, and errors due to context (timeout, etc), which are not encrypted.

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 4]
```diff
@@ -492,43 +598,21 @@ private boolean checkTokenInput(InputUtil.InputVal input, RoutingContext rc) {
...
+    }
 
-            final JsonObject resp = new JsonObject();
-            resp.put("mapped", mapped);
-            ResponseUtil.Success(rc, resp);
+    private void handleIdentityMapV2(RoutingContext rc) {
+        try {
+            final JsonObject obj = (JsonObject) rc.data().get("request");
+            JsonObject resp = handleIdentityMapCommon(rc, obj, v2IdService);
+            if (resp != null)
```
[the remark this answers is not in the document]
Comment (haitaottd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:614]: That only happens when there's an error in `handleIdentityMapCommon`. Error response should have been set.

**Review by tqinli — APPROVED** [review]
LGTM

**Pull request merged**